### PR TITLE
Add version custom parameter for new PostgreSQL instances

### PR DESCRIPTION
**Pull request by kwadwok15**
This changeset will allow a user to specify an approved major version when creating new PostgreSQL instances.

## Changes proposed in this pull request:
- Added a new option in the RDS broker, `version`
- Added checks for PostgreSQL and a valid version when provisioning a new instance 
- Added new unit tests for the new functionality
- Added a new smoke test for the new functionality
- Fixes string formatting issues flagged in more recent versions of Go

## Security considerations
- This update allows users to select an approved version in RDS, which may be older than the current version but still supported by AWS (AWS will automatically choose its latest available minor/patch release for the major version specified)

### CODE REVIEW

**Comment by ccostino** [comment]
Thanks for putting this up, @kwadwok15!

We were both working through this on Friday and I think we were able to get the crux of the feature in place. :-) A few things of note that we still need to iron out:

- What the data type of the option should be - we think this might have to be a `string` type instead, especially in light of the [AWS RDS Go SDK documentation](https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RDS.CreateDBInstance)
- Checking to see how we can account for taking in a major version and selecting the latest available minor and patch level release of that version
- Ensuring our check for Postgres is working (writing tests)
- Getting help with finalizing local setup to run tests locally (I couldn't remember how to do all of this and ran into some issues as well) and documenting the rest of those steps

**Comment by kwadwok15** [comment]
Updates: 

- Per AWS documentation, **EngineVersion** type should be "String"  [AWS CreateDBInstance API Reference](https://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html)
- As stated in AWS documentation, "If a major version is specified but a minor version is not, Amazon RDS defaults to a recent release of the major version you have specified." [PostgreSQL on Amazon RDS](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts)

File: services/rds/broker.go [hunk 1]
```diff
@@ -18,13 +18,20 @@ type RDSOptions struct {
...
+	Version            int64 `json:"version"`
 }
 
 func (r RDSOptions) Validate(settings *config.Settings) error {
 	if r.AllocatedStorage > settings.MaxAllocatedStorage {
 		return fmt.Errorf("Invalid storage %d; must be <= %d", r.AllocatedStorage, settings.MaxAllocatedStorage)
 	}
+
+	// this check only checks for psql version
+	if r.Version < 10 || r.Version > 12 {
```
Comment (jfredrickson5) [inline services/rds/broker.go:30]: Should we set the permissible version range somewhere in a config file or an environment variable?

File: services/rds/rdsinstance.go [hunk 2]
```diff
@@ -134,6 +134,10 @@ func (i *RDSInstance) init(uuid string,
 	// Load AWS values
 	i.DbType = plan.DbType
 	i.DbVersion = plan.DbVersion
+	if i.DbType == "postgres" && options.Version != "" {
```
Comment (jfredrickson5) [inline services/rds/rdsinstance.go:137]: Not to get too ahead of ourselves, but would it make sense to allow configurable versions for any database? I imagine if this use case comes up for Postrgres, it could come up for MySQL too... but then, it seems like Postgres versions tend to have more breaking changes than MySQL.

We do have a customer needing Postgres right now, so don't let me introduce unnecessary scope creep here!

File: services/rds/broker.go (hunk 1 above)
Comment (bengerman13) [reply services/rds/broker.go:30]: 💯

File: services/rds/rdsinstance.go (hunk 2 above)
Comment (bengerman13) [reply services/rds/rdsinstance.go:137]: yeah, I think this makes sense, but maybe we prove it out with PG, then add it to mysql later

Comment (ccostino) [reply services/rds/rdsinstance.go:137]: Yes and yes!  That was what we discussed, let's get it working first, then figure out how this works across the board. :-)

Furthermore, we have our MySQL versions pinned in the catalog, but we don't with PostgreSQL.  I'd like to investigate that too.

File: services/rds/broker.go (hunk 1 above)
Comment (ccostino) [reply services/rds/broker.go]: Agreed - we're going to shift this to being managed in the catalog in the future.

File: services/rds/rdsinstance.go (hunk 2 above)
Comment (ccostino) [reply services/rds/rdsinstance.go]: Captured in a new issue here: https://github.com/cloud-gov/aws-broker/issues/147

**Comment by ccostino** [comment]
@cloud-gov/platform-ops, this should be good to go now!  A huge thanks to @kwadwok15 for getting this together.  We paired on the final bits as we were working through the tests.

**Comment by ccostino** [comment]
The Concourse pipeline has been reset as well.

**Review by onelittlebecca — APPROVED** [review]
LGTM
